### DISCUSSION TITLE
Should we have a pipeline that validates our schema against the samples provided in the repo?

### DISCUSSION THREAD

**Discussion by nickkaczmarek** [wwt/WorkflowSchema#13] [Ideas] [answered]
We have a question in our PR template about updating the samples and I'm wondering if we could take the additional step of having a pipeline that validates our schema against the samples. I finally got `ajv` working with this command `npx ajv-cli --strict=false -s workflow-schema.json -d sample.json` and it also works for yaml so I'm thinking it might be nice to have. We could have it run on PR creation/update/etc so that it enforces the PR template rule.

Looks like we can also use `npx ajv-cli compile --strict=false -s workflow-schema.json` to validate that the schema itself is valid. Starting this discussion because I think it's a good idea, but want to gather some input before trying it out.

### Sidenote
we could probably update our schema such that we wouldn't need to use the `--strict=false` flag, but haven't explored that yet.

**Comment by Tyler-Keith-Thompson** [comment, marked as answer]
Strict is normally a good thing? I'd like to understand what it is about our current schema that strict has an issue with.

That said I am 100% in favor of having a pipeline that validates the schema and its examples. This gives everybody fast feedback if something *looks* right but isn't.

**Reply by nickkaczmarek** [reply]
I mentioned that in the PR, let me go get that link. but tl;dr it has to do with our items property being an object and not an array.

**Reply by nickkaczmarek** [reply]
[Whoops it was in the issue actually.](https://github.com/wwt/WorkflowSchema/issues/10#issuecomment-1040767998) This links to the issue and then there is a link to the docs for the cli and then that links to the spec for json schema. 🥴

**Reply by morganzellers** [reply]
> That said I am 100% in favor of having a pipeline that validates the schema and its examples. This gives everybody fast feedback if something _looks_ right but isn't.

+1
